src/sql/src/plan/statement/ddl.rs

```diff
@@ -1417,6 +1423,14 @@ pub fn plan_create_subsource(
             ingestion_id,
             external_reference,
             details,
+            // Subsources don't currently support non-default envelopes / encoding
+            data_config: SourceExportDataConfig {
```

Comment:
I didn't understand this part. The comment talks about default envelopes which I'm not sure what they are and then proceeds to set the None envelope which is a specific envelope. Can you elaborate on what we're trying to do here?

Reply:
Ah sorry the comment should be better, will update! Subsource statements only create 'source exports' for postgres/mysql and non-key-value load-gen sources which only support the `None` envelope and no encoding, which is why I'm just defaulting to these values here. This whole `plan_create_subsource` method will also be removed once we drop support for subsources entirely.